pkg/tcpip/network/ipv4/icmp.go

```diff
@@ -147,3 +147,37 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, vv buffer.V
 		received.Invalid.Increment()
 	}
 }
+
+func (e *endpoint) ParamProblem(r *stack.Route, netHeader buffer.View, vv buffer.VectorisedView) {
```

Comment:
I am confused why is this invoked in the write path?

Reply:
I think Parameter Problem Message is invoked because of ipv4 header parameter problems, so it is the right path to call from ipv4 directly. if any better path we can invoke this please suggest.